pkg/metamask/server.go

```diff
@@ -0,0 +1,36 @@
+package metamask
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/semrush/zenrpc/v2"
+	"go.uber.org/zap"
+	"log"
+	"net/http"
+	"os"
+)
+
+func RunMetaMaskService(ctx context.Context, address string) error {
+	rpc := zenrpc.NewServer(zenrpc.Options{ExposeSMD: true, AllowCORS: true})
+	rpc.Register("", MetaMask{}) // public
+	rpc.Use(zenrpc.Logger(log.New(os.Stderr, "", log.LstdFlags)))
+
+	http.Handle("/", rpc)
+
+	server := &http.Server{Addr: address, Handler: nil}
+
+	go func() {
+		err := server.Shutdown(ctx)
```

Comment:
You should wait for `ctx.Done()`. For example:
```go
<-ctx.Done()
zap.S().Info("shutting down metamask service...")
err := server.Shutdown(ctx)
// ....
```

Reply:
I shouldn't. <-ctx.Done() is already in body of function server.Shutdown(ctx)

Reply:
Nope, you should. In first lines in body of function `server.Shutdown(ctx)` you can see that server close listeners. It means that new incoming requests will be rejected.